### ISSUE TITLE
FCN8 code 

### ISSUE THREAD

**Issue by gsujan** [a514514772/DISE-Domain-Invariant-Structure-Extraction#1]
Hi 

I want to try this out with FCN 8 with VGG16 backend. Do you have some code to show how the decoder architecture changes and how you use the skip connections? If not code , do you have some documentation on the changes to the architecture. 

Thanks in advance.

**Comment by hui-po-wang** [comment]
Hi,

It seems that there are lots of open source code for FCN8 online. We empirically found that the features from lower layers of the shared encoder are most informative. The intuition behind it is that lower-level features, such as edges and surfaces, could be equally important for extracting structures or textures. You could probably start from using them as the input of an private encoder.

Thanks.

**Comment by gsujan** [comment]
Thanks. I ll try that out

**Comment by gsujansai** [comment]
Hi 
I have replaced the deeplab v2 with FCN 8 based on VGG 16 in the shared encoder. I initialize it with pretrained imagenet weights. Below is the result i get. 


![image](https://user-images.githubusercontent.com/27744000/57912232-1be01100-788a-11e9-9a95-f7f8c1195531.png)

While the segmentation seems to be working correctly, the reconstructed images are not meaningful. 

The shared encoder output dimension for FCN8 is 2, 4096, 9, 17 ( FCN8 downsamples to 1/32 size ) 
The shared encoder output dimensions for deeplab is 2, 2048, 33, 65. 

Deeplab output dimensions are 7 times that of FCN8. 

Could this be the reason for the bad reconstruction? 
Do you have any suggestions on what  might be the issue for bad reconstruction here ? 

Regards

**Comment by gsujan** [comment]
Hi
I have replaced the deeplab v2 with FCN 8 based on VGG 16 in the shared encoder. I initialize it with pretrained imagenet weights. Below is the result i get.

image
![image](https://user-images.githubusercontent.com/27744000/57912855-9fe6c880-788b-11e9-8252-181260beffe7.png)


While the segmentation seems to be working correctly, the reconstructed images are not meaningful.

The shared encoder output dimension for FCN8 is 2, 4096, 9, 17 ( FCN8 downsamples to 1/32 size )
The shared encoder output dimensions for deeplab is 2, 2048, 33, 65.

Deeplab output dimensions are 7 times that of FCN8.

Could this be the reason for the bad reconstruction?
Do you have any suggestions on what might be the issue for bad reconstruction here ?

Regards

**Comment by hui-po-wang** [comment]
Hi @gsujansai,

The small spatial resolution of feature maps may be the problem.

I would suggest you balance the weights between adversarial loss and reconstruction loss for image translation. BTW, I am not sure if it helps, but you probably can pretrain FCN8s on GTA5 and then use DISE to finetune it.

Thanks
